### DISCUSSION TITLE
How to select a device?

### DISCUSSION THREAD

**Discussion by rahulunair** [JuliaGPU/oneAPI.jl#226] [Q&A]
I have 2 Intel gpus on my machine - an igpu and a dgpu, how can i select the dgpu (second device) instead of the igpu, which is the default.

Output of versioninfo:

```bash
Binary dependencies:
- NEO_jll: 22.25.23529+0
- libigc_jll: 1.0.11378+0
- gmmlib_jll: 22.1.3+0
- SPIRV_LLVM_Translator_unified_jll: 0.2.0+0
- SPIRV_Tools_jll: 2022.1.0+0

Toolchain:
- Julia: 1.8.2
- LLVM: 13.0.1


2 devices:
- Intel(R) Graphics [0x5693]
- Intel(R) Graphics [0x46a6]
```

**Comment by maleadt** [comment]
By using the `device!` function: `device!(ZeDevice(1))` should do the trick. But I haven't verified whether everything works accordingly, since I don't have a multiGPU system (e.g., all functionality should use `device()` to query the active device).
